troubadix/runner.py

```diff
@@ -67,14 +67,22 @@ def __init__(
         excluded_plugins: List[str] = None,
         included_plugins: List[str] = None,
         debug: bool = False,
+        log_file: Path = None,
     ) -> None:
         self.plugins = Plugins(excluded_plugins, included_plugins)
         self._term = term
         self._n_jobs = n_jobs
         self.debug = debug
+        self._log_file = log_file
         self.special_tag_pattern = SpecialScriptTagPatterns()
         self.tag_pattern = ScriptTagPatterns()
 
+    def _log_append(self, message: str):
+        if not self._log_file:
+            return
+        with self._log_file.open(mode="a", encoding="utf-8") as f:
+            f.write(f"{message}\n")
+
```

Comment:
```suggestion
        if self._log_file:
            with self._log_file.open(mode="a", encoding="utf-8") as f:
                f.write(f"{message}\n")

```

This would also work.

I wonder if we should print on the console, too, if a logfile is passed.

I think I would love to implement something like [this](https://docs.python.org/3/library/argparse.html#action) for verbosity. :D

```python
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--verbose', '-v', action='count', default=0)
>>> parser.parse_args(['-vvv'])
Namespace(verbose=3)
```

Reply:
Maybe move " if self._log_file " to the _report_* functions?

Reply:
mhh. this would increase code duplication ... I think i is okayish this way

Reply:
verbose mode added